### PR TITLE
feat: add loading bars, ref code

### PR DESCRIPTION
**Pull request by LBokan**
### Description

Added loading bars and theme to the LocalStorage. Refactored code

### Type of change

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [x] Configuration
- [x] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)
- [x] Material UI components/constants/hooks have been used wherever possible

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

Add screenshots of the changed components here.

#### Screenshots/Video after change

Add screenshots with the changes applied here.

### CODE REVIEW

File: src/client/components/DeviceList/DeviceList.jsx [hunk 1]
```diff
@@ -30,113 +35,147 @@ export const DeviceList = ({
...
-              maxLogsQty={maxLogsQty}
-              isShortView={isShortView}
-            />
-          ))}
-        </Stack>
+      {!!isOpenEditDeviceSuccess && (
+        <NotificationBar
+          text={'Editing of the device was successful'}
+          typeOfBar="success"
+          setIsOpenBarOnComplete={setIsOpenEditDeviceSuccess}
```
Comment (ahodkov) [inline src/client/components/DeviceList/DeviceList.jsx]: Думаю, есть смысл сделать это не двумя отдельными компонентами, а просто сделать проверку:
{ (!!setIsOpenEditDeviceSuccess || !!setIsOpenDeleteDeviceSuccess) && (<NotificationBar text={setIsOpenEditDeviceSuccess ? 'Editing of the device was successful' : 'Deletion of the device was successful'} typeOfBar="success" setIsOpenBarOnComplete={setIsOpenEditDeviceSuccess || setIsOpenDeleteDeviceSuccess} /> }

Comment (ahodkov) [reply src/client/components/DeviceList/DeviceList.jsx]: Либо сделайте проще: 
Добавьте два useState(false) и useState("")
и в месте где проходят удаление/редактирование, просто передавайте сообщение и переводите useState в true

File: src/client/components/EditDeviceModal/EditDeviceModal.jsx [hunk 2]
```diff
@@ -194,8 +199,13 @@ export const EditDeviceModal = ({ deviceData, isOpen, onClose, onSubmit }) => {
...
 
             <Stack direction="row" justifyContent="flex-end">
-              <Button variant="contained" sx={{ height: '45px' }} type="submit">
-                Submit
+              <Button
+                variant="contained"
+                sx={{ minWidth: '100px', height: '45px' }}
+                type="submit"
+              >
+                {(isLoading && <ProgressBar size="20px" color="inherit" />) ||
```
Comment (ahodkov) [inline src/client/components/EditDeviceModal/EditDeviceModal.jsx:207]: Можно сделать не loading, а просто заблокировать кнопку disabled свойство

File: src/client/components/DeviceList/DeviceList.jsx (hunk 1 above)
Comment (LBokan) [reply src/client/components/DeviceList/DeviceList.jsx]: Добавила state для NotificationBar, если их больше одного на странице

File: src/client/components/EditDeviceModal/EditDeviceModal.jsx (hunk 2 above)
Comment (LBokan) [reply src/client/components/EditDeviceModal/EditDeviceModal.jsx:207]: Не меняла

**Review by ahodkov — APPROVED** [review]

**Pull request merged**